narwhals/_dask/dataframe.py

```diff
@@ -110,7 +110,7 @@ def collect(
             from narwhals._pandas_like.dataframe import PandasLikeDataFrame
 
             return PandasLikeDataFrame(
-                result,
+                result.rename_axis(columns=self.native.columns.name),
```

Comment:
This might be one step ahead of [dask#11874](https://github.com/dask/dask/issues/11874)